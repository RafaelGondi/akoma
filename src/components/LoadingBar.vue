<template>
	<div class="progress">
		<div
			class="
				progress-bar
				bg-success
				progress-bar-striped progress-bar-animated
			"
			style="width: 100%"
		></div>
	</div>
</template>

<script>
import { colorOptions } from '../utils/constants/colors';

const sizeOptions = ['sm', 'md', 'lg'];

export default {
	props: {
		/**
		 *	Propriedade referente ao tamanho do componente ('sm', 'md' ou 'lg')
		 *
		 */
		size: {
			type: String,
			default: 'md',
			validator: (value) => sizeOptions.indexOf(value) > -1,
		},
		/**
		*	Variante de cor do componente ('turquoise', 'green', 'blue',
			'indigo', 'violet', 'pink', 'red', 'orange', 'amber')
		*
		*/
		variant: {
			type: String,
			default: 'green',
			validator: (value) => colorOptions.indexOf(value) > -1,
		},
	},

	computed: {
		computedSizeClass() {
			return `spinner--${this.size}`;
		},
		computedColorClass() {
			return `spinner--${this.variant}`;
		},
		computedSpinnerClass() {
			return `${this.computedSizeClass} ${this.computedColorClass}`;
		},
	},
};
</script>

<style lang="scss" scoped>
@import '../assets/sass/app.scss';
</style>
