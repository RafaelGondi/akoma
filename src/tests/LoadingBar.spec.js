import { mount, createLocalVue } from "@vue/test-utils";
import LoadingBar from "../../src/components/LoadingBar.vue";

const localVue = createLocalVue();

test("Component is mounted properly", () => {
  const wrapper = mount(LoadingBar, {
    localVue,
  });
  expect(wrapper).toMatchSnapshot();
});
