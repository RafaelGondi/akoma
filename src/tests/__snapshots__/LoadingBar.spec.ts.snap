// Vitest Snapshot v1

exports[`LoadingBar > renders correctly 1`] = `
"<div data-v-cc77174b=\\"\\">
  <div data-v-cc77174b=\\"\\" class=\\"loading-bar__container\\">
    <div data-v-cc77174b=\\"\\" class=\\"loading-bar\\"></div>
  </div>
</div>"
`;
